### PR TITLE
Update yesod-auth and yesod-persistent to persistent-2.9

### PR DESCRIPTION
**Pull request by iand675**
This is an accompanying PR to fix changes from https://github.com/yesodweb/persistent/pull/812

Could use advice on the following, not sure whether to bump the patch version or the minor version.
- [x] Bumped the version number

### CODE REVIEW

**Comment by snoyberg** [comment]
It would be better to use CPP to support both the old and new version of persistent. And I'd recommend a minor version bump.

File: yesod-auth/yesod-auth.cabal [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name:            yesod-auth
-version:         1.6.3
+version:         1.7.0
```
[the remark this answers is not in the document]
Comment (iand675) [reply yesod-auth/yesod-auth.cabal]: Ah I see. I was thinking about it in terms of semantic versioning. That explains my confusion.

**Comment by iand675** [comment]
@snoyberg I believe I've addressed your feedback.

**Comment by iand675** [comment]
Although obviously the persistent one needs merging and releasing before this one has value.

**Comment by snoyberg** [comment]
Thanks. Sorry, just one more request: could you add a ChangeLog entry to each package that has changed explaining the support for persistent 2.9?

**Comment by iand675** [comment]
@snoyberg done. Thanks as always for your hard work!

**Comment by iand675** [comment]
I don't think it's strictly necessary to wait, but I don't think the persistent PR has been reviewed at all yet. So... if there are any requested interface changes there it might be problematic.

**Comment by snoyberg** [comment]
In that case, I'd say let's hold off for now. Would you mind pinging this issue when the 2.9 release happens?

**Pull request merged**